src/opm/output/eclipse/AggregateAquiferData.cpp

```diff
@@ -236,6 +236,10 @@ namespace {
                            ICAqArray&                       icaq)
         {
             using Ix = VI::IAnalyticAquiferConn::index;
+            const std::vector<int>& actnum = grid.getACTNUM();
+            if(!actnum[connection.global_index]) {
+                return;
+            }
```

Comment:
This doesn't work.  For one, this code section applies only to the integer `ICAQ` array, not the single precision (`float`) `SCAQ` or double precision `ACAQ` arrays.  In other words, the restart file will have an inconsistent structure.  Second, this only papers over the issue as the analytic aquifer objects on the simulator side are [initialised](https://github.com/OPM/opm-simulators/blob/7b833b7706b592f8728f153d4693f97d92f90150/opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp#L191) with the full set of connections.

The appropriate course of action here is to actually make the `AquiferConfig` object aware that cells have been removed&mdash;ideally by not creating the object until the set of active cells is fully known.